File: util/src/main/java/dev/openrs2/util/io/SkipOutputStream.java
package dev.openrs2.util.io;

import java.io.FilterOutputStream;
import java.io.IOException;
import java.io.OutputStream;

public final class SkipOutputStream extends FilterOutputStream {
	private long skipBytes;

	public SkipOutputStream(OutputStream out, long skipBytes) {
		super(out);
		this.skipBytes = skipBytes;
	}

	@Override
	public void write(int b) throws IOException {
		if (skipBytes == 0) {
			super.write(b);
		} else {
			skipBytes--;
		}
	}

	@Override
	public void write(byte[] b, int off, int len) throws IOException {
		if (len >= skipBytes) {
			off += skipBytes;
			len -= skipBytes;
			skipBytes = 0;
		} else {
			skipBytes -= len;
			return;
		}

		super.write(b, off, len);
	}
}

File: common/src/main/java/dev/openrs2/common/io/SkipOutputStream.kt
package dev.openrs2.common.io

import java.io.OutputStream

class SkipOutputStream(private val out: OutputStream, private var skipBytes: Long) : OutputStream() {
    override fun write(b: Int) {
        if (skipBytes == 0L) {
            out.write(b)
        } else {
            skipBytes--
        }
    }

    override fun write(b: ByteArray, off: Int, len: Int) {
        if (len > skipBytes) {
            out.write(b, off + skipBytes.toInt(), len - skipBytes.toInt())
            skipBytes = 0
        } else {
            skipBytes -= len
        }
    }

    override fun flush() {
        out.flush()
    }

    override fun close() {
        out.close()
    }
}
